File: SDKs/twilio.com/twilio_preview/1.36.0/go/pkg/models/operations/deletesyncdocument.go
package operations

import (
	"openapi/pkg/models/shared"
)

var DeleteSyncDocumentServers = []string{
	"https://preview.twilio.com",
}

type DeleteSyncDocumentPathParams struct {
	ServiceSid string `pathParam:"style=simple,explode=false,name=ServiceSid"`
	Sid        string `pathParam:"style=simple,explode=false,name=Sid"`
}

type DeleteSyncDocumentSecurity struct {
	AccountSidAuthToken shared.SchemeAccountSidAuthToken `security:"scheme,type=http,subtype=basic"`
}

type DeleteSyncDocumentRequest struct {
	ServerURL  *string
	PathParams DeleteSyncDocumentPathParams
	Security   DeleteSyncDocumentSecurity
}

type DeleteSyncDocumentResponse struct {
	ContentType string
	StatusCode  int64
}

File: SDKs/twilio.com/twilio_preview/1.37.3/typescript/src/sdk/models/operations/deletesyncdocument.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";

export const DeleteSyncDocumentServerList = [
	"https://preview.twilio.com",
] as const;


export class DeleteSyncDocumentPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=ServiceSid" })
  serviceSid: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=Sid" })
  sid: string;
}


export class DeleteSyncDocumentSecurity extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=http;subtype=basic" })
  accountSidAuthToken: shared.SchemeAccountSidAuthToken;
}


export class DeleteSyncDocumentRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  serverUrl?: string;

  @SpeakeasyMetadata()
  pathParams: DeleteSyncDocumentPathParams;

  @SpeakeasyMetadata()
  security: DeleteSyncDocumentSecurity;
}


export class DeleteSyncDocumentResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}
